include/boost/variant/variant.hpp

```diff
@@ -2452,16 +2455,16 @@ struct make_variant_over
 private: // precondition assertions
 
     BOOST_STATIC_ASSERT(( ::boost::mpl::is_sequence<Types>::value ));
+    typedef typename mpl::copy<Types, boost::mpl::front_inserter<mpl::list<> > >::type copied_sequence_t;
```

Comment:
Oops... `std::get<0>` exists only in C++17 Standard and in my plans for the `boost::variant`.

However, there is a `boost::variant::which()` function that allows you to get the index of a stored value:
```
BOOST_CHECK(boost::get<int>(c) == 1);
BOOST_CHECK(c.which() == 0);

v3 d = b;
BOOST_CHECK(boost::get<std::string>(d) == "2");
BOOST_CHECK(d.which() == 1);
```
`which()` will report different values if `make_variant_over` reverses the sequence :(